### PR TITLE
Issues 1090 1093 activity log

### LINKED ISSUE #1090: Minor bug in g.conv.actlog when reading dates
When reading dates from an activity log in `g.conv.actlog`, the function `data.table::fread` loads the date columns with `class = "IDate" "Date" `. This triggers an error in check_log as it expects `class = character`.

Propose including this line in `g.conv.actlog` to make sure that all columns are introduced as character in `check_log`:
`actlog[, -1] = apply(actlog[, -1], 2, as.character)`

### LINKED ISSUE #1093: Make qwindow2timestamp consider epoch size
Function `qwindow2timestamp` internally defined in `g.part5.definedays` does not consider the epoch size when turning `qwindow` numbers to time stamps. This can be problematic with rounding decimals, for example, for a time defined in activity log of **08:10:00**,  I get the `qwindow_value` of **8.166667**. When converting back **8.166667** to time stamp, I get **8:09:59** instead of **8:10:00**, which causes problems later on because the **8:09:59** timestamp is not available in the time series (epoch is 5 seconds).

The issue occurs here (g.part5.definedays, line 85):
```
for (bi in 1:length(breaks)) {
        if (any(grepl(breaks[bi], ts$time[fullQqq]))) {     # this fails to find the break in the time series
          breaks_i[bi] = fullQqq[grep(breaks[bi], ts$time[fullQqq])]
        }
        else {
          breaks_i[bi] = qqq[1]
        }
      }
```

I propose adding this to the qwindow2timestamp function to refine the timestamp conversion:

```
expected_S = seq(0, 60, by = epochSize)
  if (any(!S %in% expected_S)) {
    si = which(!S %in% expected_S)
    S[si] = expected_S[which.min(abs(expected_S - S[si]))]
    if (S[si] == 60) { # shift minute if S == 60
      M[si] = M[si] + 1
      S[si] = 0
    }
  }
```

### PR DESCRIPTION
**Pull request by jhmigueles**
Fixes #1090  => activity log dates are now read as character independently of the format of the date.
Fixes #1093  => activity log times are now interpreted considering the epoch size of the analysis to ensure matching of activity log times with time series times in part 5.

Checklist before merging:

- [x] Existing tests still work (check by running the test suite, e.g. from RStudio).
- [x] Added tests (if you added functionality) or fixed existing test (if you fixed a bug).
- [x] Clean code has been attempted, e.g. intuitive object names and no code redundancy.
- [ ] Documentation updated:
  - [ ] Function documentation
  - [ ] Chapter vignettes for GitHub IO
  - [ ] Vignettes for CRAN
- [x] Corresponding issue tagged in PR message. If no issue exist, please create an issue and tag it.
- [x] Updated release notes in `inst/NEWS.Rd` with a user-readable summary. Please, include references to relevant issues or PR discussions.
- [ ] Added your name to the contributors lists in the `DESCRIPTION` file, if you think you made a significant contribution.
- [ ] GGIR parameters were added/removed. If yes, please also complete checklist below.

If NEW GGIR parameter(s) were added then these NEW parameter(s) are :
- [ ] documented in `man/GGIR.Rd`
- [ ] included with a default in `R/load_params.R`
- [ ] included with value class check in `R/check_params.R`
- [ ] included in table of `vignettes/GGIRParameters.Rmd` with references to the GGIR parts the parameter is used in.
- [ ] mentioned in NEWS.Rd as NEW parameter

If GGIR parameter(s) were deprecated these parameter(s) are:
- [ ] documented as deprecated in `man/GGIR.Rd`
- [ ] removed from `R/load_params.R`
- [ ] removed from `R/check_params.R`
- [ ] removed from table in `vignettes/GGIRParameters.Rmd`
- [ ] mentioned as deprecated parameter in NEWS.Rd
- [ ] added to the list in `R/extract_params.R` with deprecated parameters such that these do not produce warnings when found in old config.csv files.

### CODE REVIEW

File: NEWS.md [hunk 1]
```diff
@@ -1,3 +1,7 @@
+# CHANGES IN GGIR VERSION 3.0-10
+
+- Day segmentation: Fix minor bugs in conversion of activity log timestamps #1090 and #1093.
```
Comment (vincentvanhees) [inline NEWS.md:3]: These should be two different items. The first one is not a bug but an enhancement. Can you rephrase as:
- Part 2 and 5: qwindow functionality enhanced to also consider fractions of minutes, `#1093`
- Part 2 and 5: Fix minor bug in activity diary loading when date format is %Y-%m-%d, `#1090`

File: R/g.part5.definedays.R [hunk 2]
```diff
@@ -5,10 +5,21 @@ g.part5.definedays = function(nightsi, wi, indjump, nightsi_bu,
...
-  qwindow2timestamp = function(qwindow) {
+  qwindow2timestamp = function(qwindow, epochSize) {
     H = floor(qwindow)
     M = floor((qwindow - H) * 60)
     S = floor((qwindow - H - M/60) * 60 * 60)
+    expected_S = seq(0, 60, by = epochSize)
+    if (any(!S %in% expected_S)) {
+      revise = which(!S %in% expected_S)
+      for (si in revise) {
+        S[si] = expected_S[which.min(abs(expected_S - S[si]))]
```
Comment (vincentvanhees) [inline R/g.part5.definedays.R:16]: Let's avoid single letter object names H M and S, and instead rename them to hour, minute, and second. single letter object names are very hard to search when trying to debug code. It is better to use slightly longer names that clarify what you are calculating.

Object `revise` is also not specific, maybe good to replace by seconds_tobe_revised or similar.

File: R/g.part5.definedays.R [hunk 3]
```diff
@@ -5,10 +5,21 @@ g.part5.definedays = function(nightsi, wi, indjump, nightsi_bu,
   Nts = nrow(ts)
   lastDay = FALSE
   # define local functions ----
-  qwindow2timestamp = function(qwindow) {
+  qwindow2timestamp = function(qwindow, epochSize) {
     H = floor(qwindow)
     M = floor((qwindow - H) * 60)
     S = floor((qwindow - H - M/60) * 60 * 60)
+    expected_S = seq(0, 60, by = epochSize)
+    if (any(!S %in% expected_S)) {
```
Comment (vincentvanhees) [inline R/g.part5.definedays.R:13]: `!S %in% expected_S` is performed twice. It would read easier if you split it up in

```
seconds_tobe_revised = which(!S %in% expected_S)
if (length(seconds_tobe_revised) > 0) {
 for (si in seconds_tobe_revised) {
 }
}
```

**Review by vincentvanhees — CHANGES_REQUESTED** [review]
Please find my suggestion below.

**Comment by jhmigueles** [comment]
Thanks @vincentvanhees, just finished applying your feedback
